src/domains/sign/api_product/sign/_base_policy.xml

```diff
@@ -1,5 +1,94 @@
+                        | ((int)HostToNetworkOrderShort((short)(host >> 16)) & 0xFFFF);
+                }
+
+                short HostToNetworkOrderShort(short host) {
+                    return (short)((((int)host & 0xFF) << 8) | (int)((host >> 8) & 0xFF));
+                }
+
+                string ipAddress = context.Request.Headers.GetValueOrDefault("x-forwarded-for","");
+                if (!string.IsNullOrEmpty(ipAddress)) {
+
+                    string[] tokens = ipAddress.Split(':');
+                    if(tokens.Length == 2) {
+                        ipAddress = tokens[0];
+                    }
+
+                    //Place IP Ranges into this list in CIDR notation
+                    // (e.g. "0.0.0.0/0") and separate with commas
+                    List<string> cidrList = new List<string>(){
+                        "10.0.0.0/8",
+                        "172.0.0.0/24",
+                        "192.168.0.0/24"
+                    };
+
+                    foreach (string cidrAddress in cidrList) {
+                        string[] cidrParts = cidrAddress.Split('/');
+                        string[] inputIPParts = ipAddress.Split('.');
+                        string[] cidrIPArray = cidrParts[0].Split('.');
+
+                        if (inputIPParts.Length == 4 && cidrIPArray.Length == 4) {
+                            byte[] inputIPBytes = new byte[] {Convert.ToByte(int.Parse(inputIPParts[0])),
+                                Convert.ToByte(int.Parse(inputIPParts[1])),
+                                Convert.ToByte(int.Parse(inputIPParts[2])),
+                                Convert.ToByte(int.Parse(inputIPParts[3])), };
+                            byte[] cidrIPBytes = new byte[] {Convert.ToByte(int.Parse(cidrIPArray[0])),
+                                Convert.ToByte(int.Parse(cidrIPArray[1])),
+                                Convert.ToByte(int.Parse(cidrIPArray[2])),
+                                Convert.ToByte(int.Parse(cidrIPArray[3])), };
+
+                            int cidrAddr = BitConverter.ToInt32(inputIPBytes,0);
+                            int ipAddr = BitConverter.ToInt32(cidrIPBytes,0);
+
+                            var host = int.Parse(cidrParts[1]);
+                            host = -1 << (32-host);
+                            var mask = HostToNetworkOrder(host);
+
+                            if (((ipAddr & mask) == (cidrAddr & mask))) {
+                                return "{{ipValidated}}";
+                            }
+                        }
+                    }
+                }
+
+                return ipAddress;
+            }</value>
+        </set-header>
+
+        <check-header
+            name="x-forwarded-for"
+            failed-check-httpcode="403"
+            failed-check-error-message="Forbidden"
+            ignore-case="true"
+        >
+            <value>10.33.215.173</value>
```

Comment:
Is this a temporary solution? I'm asking as I guess the whole CIDR filtering should be handled at application level since:

1. there should be a mapping between SubscriptionId -> CIDR(s) in order to allow only certain source IPs coming from a _specific_ subscription
2. this configuration should be dynamic and the issuer must be able to set up its CIDR(s) from the "Firma con IO" developer portal (we cannot manually update this configuration for every new subscription), so we have to let the developer portal making changes using an API

Of course, if these two requirements can be handled using the APIM that would be ideal.

Reply:
It is temporary to guarantee the filtering to out beta testers

Reply:
We are designing a solution that would allow the issuers to set their cidrs via the developer portal and, at the same time, keep the check on the APIM